src/scope.ml

```diff
@@ -415,6 +415,19 @@ let scope_rule : sig_state -> p_rule -> sym * pp_hint * rule loc = fun ss r ->
     let t = scope (M_RHS(Array.to_list map)) ss Env.empty p_rhs in
     Bindlib.unbox (Bindlib.bind_mvar vars t)
   in
+  let _, t = Bindlib.unmbind rhs in
+  let rec privacy t =
+    (* Ensure that there are no foreign private symbols used. *)
+    let h, args = Basics.get_args t in
+    begin match unfold h with
+      | Symb({sym_path; sym_mode=Priva; _},_) ->
+        if sym_path <> ss.signature.sign_path
+        then fatal p_lhs.pos "Foreign private symbol not allowed in RHS."
```

Comment:
It's not only in RHS that external private symbols are forbidden: it's every where except in the arguments of a rule left-hand side.

Reply:
The function that should be modified is find_sym, which should take an additional boolean argument saying that external private symbols are allowed or not. By default, this argument is false except when scoping the arguments of a rule left-hand side.